activationdemo/source/java/src/org/kaaproject/kaa/demo/activation/Main.java

```diff
@@ -0,0 +1,98 @@
+/**
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.activation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+/**
+ * A demo application class that use the Kaa activation API.
+ */
+public class Main {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Main.class);
+
+    public static void main(String[] args) throws IOException {
+        LOG.info("Configuration demo started");
+
+        // we need folder to save properties in it. In other case we can't create more then one KaaClient
+        if (!Files.exists(Paths.get(KaaManager.PROPERTIES_OUT_DIR))) {
+            LOG.error("No resource folder. Configuration demo cancelled");
+            return;
+        }
+
+        KaaManager manager = new KaaManager();
+        for (int i = 0; i < KaaManager.KAA_CLIENT_NUMBER; i++) {
+            try {
+                createResDir(i);
+                manager.startKaaClient(i);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        LOG.info("--= Press any key to exit =--");
+        try {
+            System.in.read();
+        } catch (IOException e) {
+            LOG.error("IOException was caught - ", e);
+        }
+
+        // Stop the Kaa client and release all the resources which were in use.
+        try {
+            deleteResDir();
```

Comment:
not sure that we need to clean up after, this procedure will remove security key files thus on every demo run we will get new endpoints.